### PR TITLE
feat: add QueryContext to RegionRequestHeader

### PR DESCRIPTION
**Pull request by Kelvinyu1117**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This refers to the issue "Pass query context down to region server service in Datanode".

## Checklist

- [ ]  I have written the necessary comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/issues/3752

### CODE REVIEW

**Comment by Kelvinyu1117** [comment]
Can you guys help to review when you have time? @killme2008  @MichaelScofield  @WenyXu

File: proto/greptime/v1/region/server.proto [hunk 1]
```diff
@@ -26,14 +26,24 @@ import "greptime/v1/ddl.proto";
 
 service Region { rpc Handle(RegionRequest) returns (RegionResponse); }
 
+message QueryContext {
```
Comment (MichaelScofield) [inline proto/greptime/v1/region/server.proto:29]: I think maybe we don't need that all fields in `QueryContext` to be in proto. I suggest:
```proto
message QueryContext {
  string current_catalog = 1;
  string current_schema = 2;
  // current_user is not needed
  timezone = 3;
  // sql_dialect is not needed
  map<string, string> extension = 4;
  // configuration_parameter is not needed
}

Comment (Kelvinyu1117) [reply proto/greptime/v1/region/server.proto:29]: Actually I thought the same, just wondering if we need to completely re-construct the context in the region server.

PR updated.

**Comment by MichaelScofield** [comment]
@Kelvinyu1117 I'm ok with the proto change. But we need to change the GreptimeDB's codes first before merging this PR.

**Comment by Kelvinyu1117** [comment]
> @Kelvinyu1117 I'm ok with the proto change. But we need to change the GreptimeDB's codes first before merging this PR.

Shouldn't the existing GreptimeDB point to the old revision?

**Comment by MichaelScofield** [comment]
@Kelvinyu1117 If another person wants to make changes to proto, he would normally fork the main branch. If proto is merged too soon, he will see some compile errors not related to his codes.

**Comment by Kelvinyu1117** [comment]
> @Kelvinyu1117 If another person wants to make changes to proto, he would normally fork the main branch. If proto is merged too soon, he will see some compile errors not related to his codes.

Understood, so should I change the URL to my fork in my GreptimeDB's branch at the moment?
What is your usual practice?

**Comment by MichaelScofield** [comment]
@Kelvinyu1117 yes

File: proto/greptime/v1/region/server.proto [hunk 2]
```diff
@@ -26,14 +26,21 @@ import "greptime/v1/ddl.proto";
...
+  string current_catalog = 1;
+  string current_schema = 2;
+  string timezone = 4;
+  map<string, string> extension = 5;
```
Comment (evenyag) [inline proto/greptime/v1/region/server.proto:33]: Rename to extensions
```suggestion
  map<string, string> extensions = 5;
```

**Review by evenyag — APPROVED** [review]

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**